src/tokenserver/extractors.rs

```diff
@@ -62,44 +109,162 @@ impl FromRequest for TokenserverRequest {
     }
 }
 
+impl FromRequest for Box<dyn Db> {
+    type Config = ();
+    type Error = Error;
+    type Future = LocalBoxFuture<'static, Result<Self, Self::Error>>;
+
+    fn from_request(req: &HttpRequest, _payload: &mut Payload) -> Self::Future {
+        let req = req.clone();
+
+        Box::pin(async move {
+            let state = get_server_state(&req)?;
+            let tokenserver_state = state.tokenserver_state.as_ref().unwrap();
+            let db = tokenserver_state.db_pool.get().map_err(|_| {
+                error!("⚠️ Could not acquire database connection");
+
+                TokenserverError::internal_error()
+            })?;
+
+            Ok(db)
+        })
+    }
+}
+
 impl FromRequest for TokenData {
     type Config = ();
     type Error = Error;
     type Future = LocalBoxFuture<'static, Result<Self, Self::Error>>;
 
-    fn from_request(req: &HttpRequest, payload: &mut Payload) -> Self::Future {
+    fn from_request(req: &HttpRequest, _payload: &mut Payload) -> Self::Future {
         let req = req.clone();
-        let mut payload = payload.take();
 
         Box::pin(async move {
-            let auth = BearerAuth::from_request(&req, &mut payload).await?;
-            let state = match req.app_data::<Data<ServerState>>() {
-                Some(s) => s,
-                None => {
-                    error!("⚠️ Could not load the app state");
-                    return Err(ValidationErrorKind::FromDetails(
-                        "Internal error".to_owned(),
-                        RequestErrorLocation::Unknown,
-                        Some("state".to_owned()),
-                        None,
-                    )
-                    .into());
+            // The request must have a valid Authorization header
+            let authorization_header = req
```

Comment:
These checks are not strictly necessary, but I wanted the error handling to match the logic [here](https://github.com/mozilla-services/tokenserver/blob/master/tokenserver/views.py#L116)